File: 3859-maximum-product-of-two-digits/3859-maximum-product-of-two-digits.js
/**
 * @param {number} n
 * @return {number}
 */
var maxProduct = function(num) {
    let n = 0;
    let m = 0;
    for (; num > 0; ) {
        let r = num % 10;
        if (r >= n) {
            m = n;
            n = r;
        }
        else if (r > m)
            m = r;
        num = Math.floor(num / 10);
    }
    return n * m;
};

File: 3859-maximum-product-of-two-digits/3859-maximum-product-of-two-digits.java
class Solution {
    public int maxProduct(int num) {
        int n = 0, m = 0;
        for (; num > 0; ) {
            int r = num % 10;
            if (r >= n) {
                m = n;
                n = r;
            }
            else if (r > m)
                m = r;
            num /= 10;
        }
        return n * m;
    }
}